core/dbt/context/runtime.py

```diff
@@ -60,6 +60,30 @@ def do_ref(*args):
     return do_ref
 
 
+def source(db_wrapper, model, config, manifest):
+    current_project = config.project_name
+
+    def do_source(source_name, table_name):
+        target_source = ParserUtils.resolve_source(
+            manifest,
+            source_name,
+            table_name,
+            current_project,
+            model.get('package_name')
+        )
+
+        if target_source is None:
+            dbt.exceptions.ref_target_not_found(
```

Comment:
oh yeah, this is an oversight - I implemented `source_target_not_found` elsewhere because I agree it's a terrible error message, I guess I missed this spot.